test/swe2d/test_anisotropic.py

```diff
@@ -137,21 +133,15 @@ def family(request):
 
 def test_sipg(family):
     options = {'element_family': family}
-    snes_it_auto = run(use_automatic_sipg_parameter=True, **options)
-    try:
-        snes_it_default = run(use_automatic_sipg_parameter=False, **options)
-        msg = "auto-sipg: snes iterations was not reduced, expected {:d} < {:d}"
-        assert snes_it_auto < snes_it_default, msg.format(snes_it_auto, snes_it_default)
-        msg = "auto-sipg: snes iterations reduced from {:d} to {:d} for {:s} PASSED"
-        print_output(msg.format(snes_it_default, snes_it_auto, family))
-    except ConvergenceError:
-        msg = "auto-sipg: snes converged where it diverged under default for {:s} PASSED"
-        print_output(msg.format(family))
+    snes_it = run(**options)
+    expected = 3
+    msg = f'snes iterations exceed expected: {snes_it} > {expected}'
+    assert snes_it <= expected, msg
```

Comment:
If I remember correctly, the elements in this mesh are only moderately anisotropic. It would be good to extend the test for some extreme cases (such as you get in r-adaptation), but I can do this in a separate PR.

Reply:
Yeah, sounds like a good idea!